src/mcmc_gibbs.cpp

```diff
@@ -0,0 +1,138 @@
+#include "mcmc_gibbs.h"
+
+/*
+ * Helper function to iterate over constraints and apply them
+ */
+double add_constraint(const std::string& name, List constraints,
+                      IntegerVector districts, NumericVector &psi_vec,
+                      std::function<double(List, int)> fn_constr) {
+    if (!constraints.containsElementNamed(name.c_str())) return 0;
+
+    List constr = constraints[name];
+    double val = 0;
+    double psi = 0.0;
+    for (int i = 0; i < constr.size(); i++) {
+        List constr_inst = constr[i];
+        double strength = constr_inst["strength"];
+        if (strength != 0) {
+            for (int dist = 0; dist < districts.size(); dist++) {
```

Comment:
with the type change we can then make this `for (auto dist : districts)` and change line 19 to `fn_constr(constr_inst, dist)`.

Reply:
Also https://github.com/alarm-redist/redist/pull/121/commits/ad2dba5378ce1279b407aeb00ef6540d215e0876